lib/puppet/functions/hiera_aws_sm.rb

```diff
@@ -166,6 +195,64 @@ def secret_key_name(key, options)
     end
   end
 
+  def secret_in_cache(key, options)
+    if options['cache_ttl'] > 0
+      file_cache_exist(options)
+      secrets = File.readlines(options['cache_file']).map(&:chomp)
+      value = key
+      secrets.each do | secret|
+        if(value == secret)
+          return true
+        end
+      end
+      return false
+    else 
+      return false
+    end
+  end
+
+  def file_cache_exist(options)
+    two_hours_ago = DateTime.now - (options['cache_ttl']/24.0)
+
+    if(File.exist?("#{options['cache_file']}"))
+      if (two_hours_ago.strftime( "%Y-%m-%d %H:%M:%S" ) > File.ctime("#{options['cache_file']}").strftime( "%Y-%m-%d %H:%M:%S" ))
+         generate_cache(options['cache_file'], options['prefixes'])
+      end
+    else
+         generate_cache(options['cache_file'], options['prefixes'])
+    end
+  end
+
+  def generate_cache(path, prefixes)
+    client = Aws::SecretsManager::Client.new(
+      region: 'eu-west-1',
```

Comment:
We are passing the region from hiera.yaml so you don't need to hardcode this region there